containers/ddev-dbserver/Dockerfile

```diff
@@ -33,32 +40,60 @@ RUN set -x; if ( command -v mariadbd && ! command -v mysqld ); then \
+RUN mkdir -p /var/lib/apt/lists/partial
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" bzip2 curl gnupg2 less lsb-release pv tzdata vim-tiny wget
+
+RUN <<EOF
+    set -eu -o pipefail
+    arch=$(dpkg --print-architecture)
+    set -x;
+    if ( ! command -v xtrabackup && ! command -v mariabackup ); then
+        curl -o /tmp/percona_release_latest.deb --fail -sSL ${PERCONA_SETUP_URL}
+        apt-get install -y /tmp/percona_release_latest.deb
+        rm /tmp/percona_release_latest.deb
+
+        # Use http when connecting. This only affects MariaDB 5.5
+        if [ "$(lsb_release -i -s)" = "Ubuntu" ] && "$(lsb_release -r -s)" <= "16.04" ]; then sed -i s/https:/http:/g /etc/apt/sources.list.d/percona.list; fi
+
+        case ${DB_MAJOR_VERSION} in
+            "8.0")
+                percona-release setup pxb80
+                apt-get update && apt-get -qq install -y percona-xtrabackup-80
+                ;;
+
+            "8.4")
+                percona-release setup pxb84lts
+                apt-get update && apt-get -qq install -y percona-xtrabackup-84
+                ;;
+            # Older versions of mysql and a few mariadb use pxb24
+            *)
+                percona-release setup pxb24
+                apt-get update && apt install -qq -y percona-xtrabackup-24
+                ;;
+        esac
+    fi
+EOF
 
-RUN apt-get -qq update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" bzip2 curl gnupg2 less lsb-release pv tzdata vim wget
-
-# If on 14.04 Ubuntu the percona repositories won't allow TLS apparently, so
-# Use http when connecting. This currently only affects MariaDB 5.5
-RUN set -x; if ( ! command -v xtrabackup && ! command -v mariabackup && [ $(arch) != "aarch64" ] ); then \
-    curl --fail -sSL https://repo.percona.com/apt/percona-release_latest.$(lsb_release -sc)_all.deb -o percona-release_latest.$(lsb_release -sc)_all.deb; \
-    dpkg -i percona-release_latest.$(lsb_release -sc)_all.deb >/dev/null; \
-    rm percona-release*.deb ; \
-    percona_setup=pxb24; \
-    xtrabackup_version=percona-xtrabackup-24 ; \
-    if [ "$(lsb_release -i -s)" = "Ubuntu" ] && "$(lsb_release -r -s)" <= "16.04" ]; then sed -i s/https:/http:/g /etc/apt/sources.list.d/percona.list; fi; \
-    if [ "$DB_VERSION" = "8.0" ]; then xtrabackup_version=percona-xtrabackup-80; percona_setup=pxb80; fi ; \
-    percona-release setup ${percona_setup}; \
-    apt-get -qq update && apt-get -qq install -y ${xtrabackup_version} >/dev/null ; \
-fi
 RUN apt-get -qq autoclean
 
 RUN rm -rf /var/lib/mysql /etc/mysql
 RUN mkdir -p /var/lib/mysql
 
 ADD files /
 
+# On bitnami-derived images, remove their default config so we get ours
+RUN <<BITNAMI_EOF
+if [ -d /opt/bitnami/mysql ]; then
+    rm -rf /opt/bitnami/mysql/conf
+fi
+BITNAMI_EOF
+
+ENV MYSQL_UNIX_PORT=/var/tmp/mysql.sock
```

Comment:
This seems to have been a thing through a long part of mysql's history, https://dev.mysql.com/doc/refman/5.7/en/environment-variables.html (shows in 8.0 and 8.4)